roles/common/tasks/config_validations.yml

```diff
@@ -206,3 +206,26 @@
     fail_msg: "OAuth is not supported on FIPS enabled cluster yet. Please disable FIPS or OAuth to setup the cluster."
   when: auth_mode == 'oauth' or auth_mode == 'ldap_with_oauth'
   tags: validate
+
+- assert:
+    that:
+      - hostvars[item]['kafka_controller_sasl_protocol'].replace(" ", "").split(',') | length > 1
+      - hostvars[item]['kafka_controller_sasl_protocol'].replace(" ", "").split(',')[0] not in ['scram', 'scram256']
+    fail_msg: "kafka_controller_sasl_protocol in kafka controller cannot have first value as scram or scram256 since inter-controller communication via SCRAM is not supported. Please use other protocols along with it."
+  loop: "{{ groups['kafka_controller'] }}"
+  when:
+    - ('kafka_controller' in groups.keys() and groups['kafka_controller'] | length > 1) | bool
```

Comment:
what does groups['kafka_controller'] > 1 mean here?

Reply:
Since controller-controller communication via SCRAM is not supported which is not an issue if there is just one controller. But as discussed offline we will not allow it even with single controller.